File: insertion_sort.java
import java.util.Scanner;

public class insertion_sort{
    public static void main(String [] args){
        Scanner scn = new Scanner(System.in);
        System.out.println("Please enter size of the Array ");
        int n = scn.nextInt(); 
        int [] arr = new int [n];
        System.out.println("Please enter elements of Array ");
        for (int i = 0; i< n; i++){
            arr[i] = scn.nextInt();
        }
        scn.close();
        insertionsort(arr);
    }

    public static void insertionsort(int [] arr){
        for (int i = 1; i<arr.length; i++){
            int current = arr[i];
            int j = i-1;
            while(j>=0 && arr[j]>current){
                arr[j+1] = arr[j];
                j--;
            }
            arr[j+1] = current;
        }
        System.out.println("Sorted Insertion array ");
        for (int i = 0; i<arr.length; i++){ // printing array
            System.out.println(arr[i]);
        }
    }
}

File: insertion_sort.py
def insertionsort(arr):
    for i in range(1, len(arr)):
        current = arr[i]
        j = i-1
        while arr[j] > current and j>=0:
            arr[j+1] = arr[j]
            j-=1
        
        arr[j+1] = current
    return arr


print("Please enter size of the Array ")
n = int(input())
arr = [None]*n
print("Please enter elements of Array ")
for i in range(n):
    arr[i] = int(input())
print("Sorted Insertion array ")
print(insertionsort(arr))